src/sugar.hpp

```diff
@@ -39,6 +40,7 @@ using StringDoubleVector = std::vector<std::pair<std::string, double>>;
 using DoublePair = std::pair<double, double>;
 using SizePair = std::pair<size_t, size_t>;
 using SizeOptionVector = std::vector<std::optional<size_t>>;
+using NodeAdditionResult = std::tuple<SizeVector, SizeVector, SizeVector, SizeVector>;
```

Comment:
`sugar.hpp` is really for universally-useful types. Let's put this somewhere a little more local.

And I'm sorry I didn't make this clearer before, but I was hoping to make this an actual class type rather than just renaming a tuple. That will make this code a little more durable if we want to encapsulate a little functionality. However, if you don't want to do that I don't mind at all. 😄

Reply:
Dave and I talked, and we think that the vectors are all loosely related, and so pulling them into a single class doesn't make much sense. We did pull it out from sugar.hpp and into the class scope though!

Reply:
Actually, disregard the above. We ended up deciding to make NodeAdditionResult a class.